### ISSUE TITLE
Warning about closed plugins

### ISSUE THREAD

**Issue by Zodiac1978** [roytanck/rt-plugin-report#51]
After checking a website with a closed plugin I was wondering that plugin report did not warn me about it.

@roytanck Does the WordPress.org Plugin API sends information about closed plugins?

**Comment by roytanck** [comment]
Interesting. This is something I'll have to look into. Would be nice if we could somehow detect this and show a warning.

**Comment by roytanck** [comment]
It looks like there's no discernible difference (in the API output) between a plugin that was closed and one that never existed. This means that the only way to determine a plugin closure is to have our plugin keep track of this. Currently (other than caching) we don't store any historical data. This means we can't detect "it was there last time, now it's gone" scenarios.

If we did implement this (by keeping at minimum one 'is in repo' boolean per installed plugin), this data would still not be available on new installs, or after migrations where the database is not copied.

**Comment by roytanck** [comment]
That might indeed work. We could even "borrow" that plugin's function that performs the check. It would increase the number of external calls we do, since you'd need to check this for all plugins not found in the repo.

There is a small chance of name collisions, because a non-repo plugin might have a slug that matches in SVN. I think most plugin authors will check if their new plugin's name exists in te repo (even if they're not immediately planning to publish there). I doubt they'll check SVN.

**Comment by roytanck** [comment]
I've created a feature branch for this issue:
https://github.com/roytanck/rt-plugin-report/tree/feature/detect-closed-plugins

The code appears to work well, but does cause a noticeable delay for plugins that are not found in the repository.

@Zodiac1978 If you could test this with the plugins that you mentioned on Twitter , that would be helpful.

**Comment by Zodiac1978** [comment]
I have tested the feature branch and haven't seen any dramaticdelay for the plugin which is closed (I've already deleted one of them).

The plugin report page is just opened if I want to check the plugins, so I don't think that this small delay would be problematic. Adding those infos to the plugins table on the plugins page (like other plugins do) creates a noticeable delay for the plugin page and would be problematic, but in our case I think it is totally fine.
